### ISSUE TITLE
multiple env files support

### ISSUE THREAD

**Issue by hartmark** [mattdavis90/immich-stacker#7]
Pixel 8 seems to have multiple formats for raw.

I have seen these two variants now:
PXL_20241030_145010658.RAW-01.COVER.jpg
PXL_20241030_145010658.RAW-02.ORIGINAL.dng

and:
PXL_20241030_143645718.RAW-01.MP.COVER.jpg
PXL_20241030_143645718.RAW-02.ORIGINAL.dng

And for my old Pixel 6 it generated images like this:
PXL_20230829_215158543.dng
PXL_20230829_215158543.jpg

So I need to have three different .env files and swap them around.

```
% head .env*      
==> .env <==
IMMICH_LOG_LEVEL=DEBUG
IMMICH_API_KEY=*****
IMMICH_ENDPOINT=*****/api
IMMICH_MATCH=(PXL.*)\.(jpg|dng)$
IMMICH_PARENT=(PXL.*)\.jpg$

==> .env.mp <==
IMMICH_LOG_LEVEL=DEBUG
IMMICH_API_KEY=*****
IMMICH_ENDPOINT=*****/api
IMMICH_MATCH=\.(RAW\-01\.MP\.COVER\.jpg|RAW\-02\.ORIGINAL\.dng)$
IMMICH_PARENT=\.jpg$

==> .env.pixel8 <==
IMMICH_LOG_LEVEL=DEBUG
IMMICH_API_KEY=*****
IMMICH_ENDPOINT=*****/api
IMMICH_MATCH=\.(RAW\-01\.COVER\.jpg|RAW\-02\.ORIGINAL\.dng)$
IMMICH_PARENT=\.jpg$

```

PS. It would be nice to have a --dry-run parameter that would just print out what it will do for easier debugging

**Comment by hartmark** [comment]
How is the regular expression working, the one for pixel 6 didn't match any images.

I want it to just search for images starting with "PXL"

**Comment by mattdavis90** [comment]
Hi,

I just pushed a `v1.4.0` release that has an `IMMICH_READ_ONLY` flag that you can set it the environment.

Supporting multiple envs is possible but I'm nervous about how they may conflict with each other and just cause chaos. This project is mostly a stop-gap solution waiting on the Immich devs to implement the feature as a 1st party into Immich so I'm slightly reluctant to incorporate complex business logic that they're likely already thinking about.

I've never tried to have both a prefix and suffix regex. The regex works by first matching image filenames, then removing the match groups from the filename, the filenames are then grouped up based on the new filenames, then whichever name would have originally matched the "parent" regex is marked as parent. Hopefully that makes sense.

I'm pretty certain you don't need the `PXL_` prefix, I think the following should work
```env
IMMICH_MATCH=\.((RAW-(01\.(MP\.)?COVER\.jpg|02\.ORIGINAL\.dng))|jpg|dng)$
IMMICH_PARENT=\.jpg$
```

or possibly even

```env
IMMICH_MATCH=\.(.*(jpg|dng))$
IMMICH_PARENT=\.jpg$
```

**Comment by hartmark** [comment]
I realized you can mix settings in .env and just set them as ordinary environment variables in the shell.

IMMICH_MATCH='\.(jpg|dng)$' IMMICH_PARENT='\.jpg$' IMMICH_READ_ONLY=true ./immich-stacker

So now my .env just contains the  serverr settings
